### ISSUE TITLE
A problem about Log norm loss 

### ISSUE THREAD

**Issue by vanetspark** [yl4579/StarGANv2-VC#44]
In your paper , the Norm consistency loss is claimed as  "  the absolute column-sum norm for a mel-spectrogram X with N mels and T frames at the t th frame, "  

but in the losses.py , it is calculated by :

def log_norm(x, mean=-4, std=4, dim=2):
    """
    normalized log mel -> mel -> norm -> log(norm)
    """
    x = torch.log(torch.exp(x * std + mean).norm(dim=dim))
    return x

              # norm consistency loss
              x_fake_norm = log_norm(mc_fake)
              x_real_norm = log_norm(mc_src)
              loss_norm = ((torch.nn.ReLU()(torch.abs(x_fake_norm - x_real_norm) -0.5)) ** 2).mean()

In my own experiment, (it has a different dataset , which is not VCTK),
I found that sometimes it will be  Inf . 

please tell me why it is not like this : 

loss_norm = torch.cumsum(x_fake - x_real, 2)  / (x.shape[-1])

**Comment by yl4579** [comment]
In the paper, there was an absolute value in equation 7, so you do need to use `torch.sum(torch.abs(x_fake - x_real), 2) / (x.shape[-1])` at least. I don't understand why you would use cumsum, because in the paper the definition of the norm is not cumulative. 

The reason I calculate the norm loss in log scale instead of linear scale is we need to normalize the melspectrogram before calculating the norm, and the melspectrograms are originally computed in log scale. However, since both exponential and logarithm functions are monotonic, using log or not does not affect the minimizers. It is possible that using log results in infinity if the norm is very small (as log goes to negative infinity when the input approaches 0). To make it numerically more stable, you may use

```python
def log_norm(x, mean=-4, std=4, dim=2):
"""
normalized log mel -> mel -> norm -> log(norm)
"""
x = (x * std + mean).norm(dim=dim)
return x
```
 instead.

**Comment by vanetspark** [comment]
Sorry , I  uncorrectly understood the equation (7) due to my short study time  in Speech deep learning .
Thank you , I will try to make the code numerically more stable .
